src/dodal/common/maths.py

```diff
@@ -0,0 +1,50 @@
+from typing import Tuple
+
+import numpy as np
+
+
+def step_to_num(start: float, stop: float, step: float) -> Tuple[float, float, int]:
+    """
+    Standard handling for converting from start, stop, step to start, stop, num
+    Forces step to be same direction as length
+    Includes a final point if it is within 1% of the end point (consistent with GDA)
```

Comment:
Nit: In general I've tried to avoid comments mentioning GDA as it seems cargocult-y the decision should stand on it's own, not reference GDA

Reply:
Adjusted to spell out the reason, may the planes return soon